sdk/test-utils/recorder-new/src/core-v1-recorder.ts

```diff
@@ -3,30 +3,67 @@
 
 import { HttpClient, HttpOperationResponse } from "@azure/core-http";
 import { DefaultHttpClient, WebResourceLike } from "@azure/core-http";
-import { isPlaybackMode, isRecordMode } from "@azure-tools/test-recorder";
-import { TestProxyHttpClient } from "./core-v2-recorder";
+import { isLiveMode } from "@azure-tools/test-recorder";
+import { RecorderClient } from "./core-v2-recorder";
 import { Test } from "mocha";
+import { ensureExistence, RecorderError } from "./utils/utils";
+import { env } from "./utils/env";
+import { RecorderRequestModifier } from "./recorderRequestModifier";
 
 /**
- * This client manages the recorder life cycle and interacts with the proxy-tool to do the recording,
- * eventually save them in record mode and playing them back in playback mode.
+ * This client modifies the sendRequest to redirect the requests to the proxy tool instead of directly going to the service.
+ * This client is supposed to be passed as the httpClient for the SDKs based on core-http.
  *
- * This client is meant for the core-v1 SDKs(depending on core-http) and
- * is supposed to be passed as the httpClient option in the options bag of your client.
+ * @export
+ * @class TestProxyHttpClientCoreV1
+ * @implements {HttpClient}
  */
-export class TestProxyHttpClientCoreV1 extends TestProxyHttpClient {
-  public httpClientCoreV1: HttpClient;
-  constructor(testContext?: Test) {
-    super(testContext);
-    this.httpClientCoreV1 = new DefaultHttpClient();
+export class TestProxyHttpClientCoreV1 implements HttpClient {
+  private _httpClient: HttpClient;
+  constructor(
+    private client: {
+      requestModifier?: RecorderRequestModifier;
+      getRecordingId: () => string | undefined;
+    }
+  ) {
+    this._httpClient = new DefaultHttpClient();
   }
 
   async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
-    if (isPlaybackMode() || isRecordMode()) {
-      if (this.recordingId) {
-        this.redirectRequest(request);
+    // If check needed because we only modify requests in record/playback modes.
+    if (
+      !isLiveMode() &&
+      ensureExistence(
```

Comment:
what about moving all of the property existing verification calls into a single “prepare sendRequest” method? then it could be very scoped and the `ensureExistence` function could go away.

Reply:
Eh, IMO the best design is one where the property is definitely-assigned and we don't have to make assertions about its existence.

Reply:
removed mode param

Reply:
Keeping the rest two params, will update in a separate PR if you feel strongly.